src/rollup/index.ts

```diff
@@ -427,7 +430,17 @@ function normalizeOutputOptions(
 	if (mergedOptions.optionError) throw new Error(mergedOptions.optionError);
 
 	// now outputOptions is an array, but rollup.rollup API doesn't support arrays
-	const outputOptions = mergedOptions.outputOptions[0];
+	const mergedOutputOptions = mergedOptions.outputOptions[0];
+	const outputOptionsReducer = (outputOptions: OutputOptions, result: OutputOptions) => {
+		if (!result) return outputOptions;
+
+		return {...outputOptions, ...result};
```

Comment:
The rest looks really good, this is the only thing giving me pause. I am not sure we should be merging the output options; at least when compared to the options hook that only replaces options. Though I would mostly argue for consistency here.

Also, this makes it impossible to remove options (though of course you could set them to `undefined`), which could be confusing for some options where `undefined` and `false` behave differently (e.g. `output.interop`).
And lastly, this is only a shallow merge which may or may not be what people expect for object valued options (there are only few, e.g. `output.amd`).

Would you be ok with a change or do you have strong arguments for merging?

Reply:
No, I don't have any strong arguments. TBH I also had some doubts about it, yet it seemed like a more elegant solution to me. However you made good point about removing options. I simplified the code to just replace options.